src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -385,53 +446,38 @@ CheckAppPolicyResults PolicyManagerImpl::CheckPermissionsChanges(
 void PolicyManagerImpl::ProcessAppPolicyCheckResults(
     const CheckAppPolicyResults& results,
     const policy_table::ApplicationPolicies& app_policies) {
-  CheckAppPolicyResults::const_iterator it_results = results.begin();
-
-  for (; results.end() != it_results; ++it_results) {
-    const policy_table::ApplicationPolicies::const_iterator app_policy =
-        app_policies.find(it_results->first);
+  policy_table::ApplicationPolicies::const_iterator iter = app_policies.begin();
+  policy_table::ApplicationPolicies::const_iterator iter_end =
+      app_policies.end();
 
-    if (app_policies.end() == app_policy) {
+  for (; iter != iter_end; ++iter) {
+    if (IsPredefinedApp(*iter)) {
       continue;
     }
 
-    if (IsPredefinedApp(*app_policy)) {
-      continue;
-    }
+    AppPolicyResultsChecker checker(iter);
+    std::for_each(results.begin(), results.end(), std::ref(checker));
```

Comment:
@AKalinich-Luxoft AFAIK `std::ref` is C++11 feature

Reply:
@AByzhynar 
I don't see any sense to avoid of using C++11 as the appropriate proposal was accepted already.

Reply:
@dev-gh I don't know about such acception

Reply:
@AByzhynar this code was removed, please review 89ceac249b54ec02c042865b27f4af8b2363d450